### DISCUSSION TITLE
Support aggregation for filter addition

### DISCUSSION THREAD

**Discussion by LeaVerou** [11ty/eleventy#4061] [Enhancement Queue]
This is a super low priority nice-to-have QoL addition, but thought I should probably open a new issue than try to stuff it into an "Oh and by the way" in #3386.

Most of my 11ty projects have an import like:

```js
import * as filters from "./filters/index.js";
```

and a mass-add statement in the config:

```js
for (let name in filters) {
	config.addFilter(name, filters[name]);
}
```

It would be nice if `addFilter()` supported aggregation: if you pass a whole object, simply add all values that are functions as filters, using their keys as names.
This would turn my little snippet into a single line:

```js
config.addFilter(filters);
```

Or, you could use a separate name:

```js
config.addFilters(filters);
```

And even for those defining filters inline in their config file, it would make the definitions far more readable:

```js
config.addFilters({
	foo () {
		/* ... */
	},
	bar () {
		/* ... */
	},
	// ...
});
```

**Comment by nhoizey** [comment]
It would be great indeed.

I would in addition love a `config.addFiltersGlob("**/filters/*.js")` that would add filters defined in multiple files. (even better if the parameter can be an array of globs).

It would ease a lot what I've done there: 😅

https://github.com/nhoizey/eleventy-plugin-pack11ty/blob/main/eleventy-plugin-pack11ty.js#L67-L94

**Comment by zachleat** [comment]
> By default this repo closes enhancement request issues and places them in a queue for folks to upvote. We do this to avoid a large build-up of open issues. We got this idea from the [lodash project](https://twitter.com/samselikoff/status/991395669016436736).

> Don’t forget to upvote the first comment with 👍 to register your vote! You can view [the current enhancement backlog](https://github.com/11ty/eleventy/issues?utf8=%E2%9C%93&q=label%3Aneeds-votes+sort%3Areactions-%2B1-desc+).
